pkg/scheduler/objects/application_state.go

```diff
@@ -142,6 +153,13 @@ func NewAppState() *fsm.FSM {
 				} else {
 					app.OnStateChange(event, "")
 				}
+				eventDetails, ok := stateEvents[event.Dst]
+				if !ok {
+					log.Log(log.SchedFSM).Error("event details not found",
+						zap.String("state", event.Dst))
+					return
+				}
```

Comment:
We are sending events based on certain cases. Which means, there is no need to send event while entering certain state. So, do we really need to log this as error? May be debug in case if there is no need to send event

Reply:
Ideally we want to send out an event for every state except New (because that's the initial state and it carries no extra info). So right now if it's missing it means a bug, that's why I dediced to log as an error. Let's wait for Wilfred's/Craig's input.

Reply:
Ok. Expired, Resuming ?

Reply:
Added.